### PR TITLE
Reactive Profile Deployments (Proxies, partially)

### PR DESCRIPTION
**Pull request by rryter**
Use the latest version of `lsp-factory.js` to deploy an `LSP3UniversalProfile`

### CODE REVIEW

File: src/components/profile/profile-deploy/ProfileDeploy.vue [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          <a
+            v-if="deploymentEvent?.receipt"
+            :href="`https://blockscout.com/lukso/l14/tx/${deploymentEvent?.receipt?.transactionHash}/internal-transactions`"
+            class="button is-small mb-1"
+          >
+            {{ deploymentEvent?.receipt?.transactionHash.substring(0, 16) }}...
+          </a>
+          <a
+            v-if="deploymentEvent?.transaction"
+            :href="`https://blockscout.com/lukso/l14/tx/${deploymentEvent?.receipt?.transactionHash}/internal-transactions`"
```
Comment (Hugoo) [inline src/components/profile/profile-deploy/ProfileDeploy.vue:62]: so you can also reuse it here

File: src/components/wallet/WalletBalance.vue [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { getBalanceFormatted, shortAddress } from "@/helpers/ethers";
+export default defineComponent({
+  name: "WalletBalance",
+  props: {
+    msg: String,
+  },
+  data: () => {
+    return {
+      balance: 0 as any,
+      address: "0x214be121bB52e6909c5158579b3458f8760f1b2f",
```
Comment (Hugoo) [inline src/components/wallet/WalletBalance.vue:19]: what is this address?

**Review by Hugoo — APPROVED** [review]
only the private key and a few constant which may be put elsewhere?
looks cool !

File: src/components/profile/profile-detail/profile-detail.component.ts [hunk 3]
```diff
@@ -3,24 +3,32 @@ import { request } from "graphql-request";
...
       uploadTarget: DEFAULT_IPFS_URL,
     };
   },
-  created() {
+  async created() {
     // fetch the data when the view is created and the data is
     // already being observed
+    const result = await getSigner();
+    if (this.$route.params.address) {
+      this.account = new LSP3Account__factory(result.signer).attach(
```
[the remark this answers is not in the document]
Comment (rryter) [reply src/components/profile/profile-detail/profile-detail.component.ts:28]: That's a convention coming from typechain

File: src/services/provider.service.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    console.log(process.env);
+    provider = new ethers.providers.JsonRpcProvider(
+      process.env.VUE_APP_JSON_RPC_PROVIDER
+    );
+    // provider = new ethers.providers.Web3Provider((window as any).ethereum);
+  }
+
+  // const signer = await provider.getSigner();
+  const signer = new ethers.Wallet(
+    "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
```
[the remark this answers is not in the document]
Comment (rryter) [reply src/services/provider.service.ts:26]: It's a private key that's coming from `npx hardhat node` it's always the same, so it's not like a "leak" or anything.

I'll move it into the .env file.

File: src/components/profile/profile-deploy/profile-deploy.component.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      this.selectedProfile = selectedProfile;
+    },
+    closeModal() {
+      this.isModalOpen = false;
+    },
+    getTypeClass(type: string) {
+      return {
+        "is-primary": type === "PROXY",
+        "is-warning": type === "CONTRACT",
+        "is-info": type === "TRANSACTION",
```
[the remark this answers is not in the document]
Comment (rryter) [reply src/components/profile/profile-deploy/profile-deploy.component.ts:79]: It already does, I'll update this.

File: src/components/profile/profile-deploy/ProfileDeploy.vue (hunk 1 above)
Comment (rryter) [reply src/components/profile/profile-deploy/ProfileDeploy.vue]: good idea

File: src/components/wallet/WalletBalance.vue (hunk 2 above)
Comment (rryter) [reply src/components/wallet/WalletBalance.vue]: it's obsolete, thanks for spotting

File: .env.production [hunk 6]
```diff
@@ -0,0 +1 @@
+VUE_APP_JSON_RPC_PROVIDER=http://rpc.l14.lukso.network/:8545
```
[the remark this answers is not in the document]
Comment (rryter) [reply .env.production:1]: I'm going to update the helm-chart to handle the env vars in the next pr. So this file will be obsolete and removed.

thanks for flagging 👍

**Pull request merged**